From c86c0b931de752742f7c13a7da6b4d90bcc06054 Mon Sep 17 00:00:00 2001
From: Xiangyi Zeng <xiangyi.zeng@linux.alibaba.com>
Date: Sun, 1 Sep 2024 01:08:44 +0800
Subject: [PATCH 227/400] drivers: wdt: drop DW wdt restart function for th1520
 soc

cherry-pick from :
Author: lw312886 <lw312886@linux.alibaba.com>
Date:   Sat Nov 6 14:55:31 2021 +0800

    configs: watchdog: light-fm: enable light pmic restart and drop DW wdt restart function

    enable light pmic wdt restart function and drop DW wdt restart function for THEAD light fm

Signed-off-by: wei.liu <lw312886@linux.alibaba.com>
Signed-off-by: Xiangyi Zeng <xiangyi.zeng@linux.alibaba.com>
Signed-off-by: Han Gao <gaohan@iscas.ac.cn>
---
 drivers/watchdog/dw_wdt.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/drivers/watchdog/dw_wdt.c b/drivers/watchdog/dw_wdt.c
index f2cfc38809cc..50ba7c98873e 100644
--- a/drivers/watchdog/dw_wdt.c
+++ b/drivers/watchdog/dw_wdt.c
@@ -298,6 +298,7 @@ static int dw_wdt_stop(struct watchdog_device *wdd)
 	return 0;
 }
 
+#ifdef CONFIG_DW_WDT_RESTART_ENA
 static int dw_wdt_restart(struct watchdog_device *wdd,
 			  unsigned long action, void *data)
 {
@@ -316,6 +317,7 @@ static int dw_wdt_restart(struct watchdog_device *wdd,
 
 	return 0;
 }
+#endif
 
 static unsigned int dw_wdt_get_timeleft(struct watchdog_device *wdd)
 {
@@ -355,7 +357,9 @@ static const struct watchdog_ops dw_wdt_ops = {
 	.set_timeout	= dw_wdt_set_timeout,
 	.set_pretimeout	= dw_wdt_set_pretimeout,
 	.get_timeleft	= dw_wdt_get_timeleft,
+#ifdef CONFIG_DW_WDT_RESTART_ENA
 	.restart	= dw_wdt_restart,
+#endif
 };
 
 static irqreturn_t dw_wdt_irq(int irq, void *devid)
-- 
2.43.0

